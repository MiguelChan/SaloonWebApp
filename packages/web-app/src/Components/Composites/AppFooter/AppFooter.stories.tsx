import React from 'react';
import {
  Story,
  Meta,
} from '@storybook/react';
import { AppFooter } from '.';

export default {
  title: 'Components/Composites/AppFoter',
  component: AppFooter,
} as Meta;

const Template: Story = () => <AppFooter />;

export const Default = Template.bind(this);