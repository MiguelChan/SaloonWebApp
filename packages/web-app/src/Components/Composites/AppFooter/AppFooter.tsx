import * as React from 'react';
import Box from '@mui/material/Box';
import Grid from '@mui/material/Grid';
import Link from '@mui/material/Link';
import Container from '@mui/material/Container';
import facebookLogo from 'Images/facebookLogo.png';
import instagramLogo from 'Images/instagramLogo.png';
import { Typography, TextField } from 'Components/Blocks';

function Copyright() {
  return (
    <React.Fragment>
      {'© '}
      <Link color="inherit" href="https://lashairbysamy.com/" target='_blank'>
        LashairBySamy
      </Link>{' '}
      {new Date().getFullYear()}
    </React.Fragment>
  );
}

const iconStyle = {
  width: 48,
  height: 48,
  display: 'flex',
  justifyContent: 'center',
  alignItems: 'center',
  backgroundColor: 'warning.main',
  mr: 1,
  '&:hover': {
    bgcolor: 'warning.dark',
  },
};

const LANGUAGES = [
  {
    code: 'en-US',
    name: 'English',
  },
  {
    code: 'fr-FR',
    name: 'Français',
  },
];

export default function AppFooter() {
  return (
    <Typography
      component="footer"
      sx={{ display: 'flex', bgcolor: 'secondary.light' }}
    >
      <Container sx={{ my: 8, display: 'flex' }}>
        <Grid container spacing={5}>
          <Grid item xs={6} sm={4} md={3}>
            <Grid
              container
              direction="column"
              justifyContent="flex-end"
              spacing={2}
              sx={{ height: 120 }}
            >
              <Grid item sx={{ display: 'flex' }}>
                <Box component="a" href="https://www.facebook.com/samyescobedoreyes" sx={iconStyle} target='_blank'>
                  <img
                    src={facebookLogo}
                    alt='Facebook'
                    width='100%'
                    height='auto'
                  />
                </Box>
                <Box
                  component="a"
                  href="https://instagram.com/lashairby"
                  sx={iconStyle}
                >
                  <img
                    src={instagramLogo}
                    alt="Instagram"
                    width='100%'
                    height='auto'
                  />
                </Box>
              </Grid>
              <Grid item>
                <Copyright />
              </Grid>
            </Grid>
          </Grid>
          <Grid item xs={6} sm={4} md={2} visibility='hidden'>
            <Typography variant="h6" marked="left" gutterBottom>
              Legal
            </Typography>
            <Box component="ul" sx={{ m: 0, listStyle: 'none', p: 0 }}>
              <Box component="li" sx={{ py: 0.5 }}>
                <Link href="/premium-themes/onepirate/terms/">Terms</Link>
              </Box>
              <Box component="li" sx={{ py: 0.5 }}>
                <Link href="/premium-themes/onepirate/privacy/">Privacy</Link>
              </Box>
            </Box>
          </Grid>
          <Grid item xs={6} sm={8} md={4} visibility='hidden'>
            <Typography variant="h6" marked="left" gutterBottom>
              Language
            </Typography>
            <TextField
              select
              size="medium"
              variant="standard"
              SelectProps={{
                native: true,
              }}
              sx={{ mt: 1, width: 150 }}
            >
              {LANGUAGES.map((language) => (
                <option value={language.code} key={language.code}>
                  {language.name}
                </option>
              ))}
            </TextField>
          </Grid>
          <Grid item>
            <Typography variant="caption">
              {'Visita nuestra redes sociales, dando click a los iconos de arriba'}
            </Typography>
          </Grid>
        </Grid>
      </Container>
    </Typography>
  );
}