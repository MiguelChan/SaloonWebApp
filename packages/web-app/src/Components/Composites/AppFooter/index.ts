export { default as AppFooter } from './AppFooter';
export * from './AppFooter';