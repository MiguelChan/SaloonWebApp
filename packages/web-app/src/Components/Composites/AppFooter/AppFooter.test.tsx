import React from 'react';
import {
  render,
  RenderResult,
} from '@testing-library/react';
import AppFooter from './AppFooter';
import { ThemeProvider } from '@mui/styles';
import {
  theme,
} from 'Utils';

describe('AppFooter', () => {
  it('Should display the AppFooter', () => {
    const { container }: RenderResult = setupComponent();
    expect(container).toMatchSnapshot();
  });

  const setupComponent = () => {
    const Component = (
      <ThemeProvider theme={theme}>
        <AppFooter />
      </ThemeProvider>
    );

    return render(Component);
  }

});