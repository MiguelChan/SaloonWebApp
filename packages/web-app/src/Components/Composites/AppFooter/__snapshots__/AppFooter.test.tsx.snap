// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppFooter Should display the AppFooter 1`] = `
<div>
  <footer
    class="MuiTypography-root MuiTypography-body1 css-gt3tvs-MuiTypography-root"
  >
    <div
      class="MuiContainer-root MuiContainer-maxWidthLg css-1wu66hi-MuiContainer-root"
    >
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-5 css-1ld3b9g-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 MuiGrid-grid-sm-4 MuiGrid-grid-md-3 css-1vm8tgf-MuiGrid-root"
        >
          <div
            class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-direction-xs-column css-1532ovy-MuiGrid-root"
          >
            <div
              class="MuiGrid-root MuiGrid-item css-15zuw48-MuiGrid-root"
            >
              <a
                class="MuiBox-root css-sh6kez"
                href="https://www.facebook.com/samyescobedoreyes"
                target="_blank"
              >
                <img
                  alt="Facebook"
                  height="auto"
                  src="facebookLogo.png"
                  width="100%"
                />
              </a>
              <a
                class="MuiBox-root css-sh6kez"
                href="https://instagram.com/lashairby"
              >
                <img
                  alt="Instagram"
                  height="auto"
                  src="instagramLogo.png"
                  width="100%"
                />
              </a>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
            >
              © 
              <a
                class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1kgfegd-MuiTypography-root-MuiLink-root"
                href="https://lashairbysamy.com/"
                target="_blank"
              >
                LashairBySamy
              </a>
               
              2022
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 MuiGrid-grid-sm-4 MuiGrid-grid-md-2 css-sd4llg-MuiGrid-root"
        >
          <h2
            class="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom css-18k87ye-MuiTypography-root"
          >
            Legal
            <span
              class="Typography-markedH6Left-4"
            />
          </h2>
          <ul
            class="MuiBox-root css-19brutj"
          >
            <li
              class="MuiBox-root css-mz9qt7"
            >
              <a
                class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1iwtne7-MuiTypography-root-MuiLink-root"
                href="/premium-themes/onepirate/terms/"
              >
                Terms
              </a>
            </li>
            <li
              class="MuiBox-root css-mz9qt7"
            >
              <a
                class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1iwtne7-MuiTypography-root-MuiLink-root"
                href="/premium-themes/onepirate/privacy/"
              >
                Privacy
              </a>
            </li>
          </ul>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 MuiGrid-grid-sm-8 MuiGrid-grid-md-4 css-x7u1a5-MuiGrid-root"
        >
          <h2
            class="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom css-18k87ye-MuiTypography-root"
          >
            Language
            <span
              class="Typography-markedH6Left-4"
            />
          </h2>
          <div
            class="MuiFormControl-root MuiTextField-root css-8icdyj-MuiFormControl-root-MuiTextField-root"
          >
            <div
              class="TextField-root-5 MuiInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-1aa5qj0-MuiInputBase-root-MuiInput-root"
            >
              <select
                aria-invalid="false"
                class="TextField-select-14 MuiNativeSelect-select MuiNativeSelect-standard TextField-input-6 TextField-inputSizeMedium-10 TextField-inputBorder-7 MuiInput-input MuiInputBase-input css-1dmqq7i-MuiNativeSelect-select-MuiInputBase-input-MuiInput-input"
                id="mui-1"
              >
                <option
                  value="en-US"
                >
                  English
                </option>
                <option
                  value="fr-FR"
                >
                  Français
                </option>
              </select>
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium TextField-selectIcon-15 MuiNativeSelect-icon MuiNativeSelect-iconStandard css-10bey84-MuiSvgIcon-root-MuiNativeSelect-icon"
                data-testid="ArrowDropDownIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7 10l5 5 5-5z"
                />
              </svg>
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
        >
          <span
            class="MuiTypography-root MuiTypography-caption css-1sn4lm3-MuiTypography-root"
          >
            Visita nuestra redes sociales, dando click a los iconos de arriba
          </span>
        </div>
      </div>
    </div>
  </footer>
</div>
`;
